### PR TITLE
Add types for the manifest.json file

### PR DESCRIPTION
**Pull request by tim-hellhake**

### CODE REVIEW

File: src/manifest.ts [hunk 1]
```diff
@@ -9,3 +9,33 @@ export interface Moziot<T> {
...
+
+export interface GatewaySpecificSettings {
+  webthings: WebthingsSettings;
+}
+
+export interface WebthingsSettings {
+  exec: string;
+  strict_min_version: string;
+  strict_max_version: string;
+  primary_type: string;
```
Comment (mrstegeman) [inline src/manifest.ts:35]: `primary_type` could be `'adapter' | 'notifier' | 'extension'`

File: src/manifest.ts [hunk 2]
```diff
@@ -9,3 +9,33 @@ export interface Moziot<T> {
...
+  gateway_specific_settings: GatewaySpecificSettings;
+  options: Options;
+}
+
+export interface GatewaySpecificSettings {
+  webthings: WebthingsSettings;
+}
+
+export interface WebthingsSettings {
+  exec: string;
```
Comment (mrstegeman) [inline src/manifest.ts:32]: `exec` is optional, since it's not required for pure extensions.

File: src/manifest.ts [hunk 3]
```diff
@@ -9,3 +9,33 @@ export interface Moziot<T> {
...
+  id: string;
+  name: string;
+  short_name: string;
+  version: string;
+  description: string;
+  homepage_url: string;
+  license: string;
+  author: string;
+  gateway_specific_settings: GatewaySpecificSettings;
+  options: Options;
```
Comment (mrstegeman) [inline src/manifest.ts:24]: `options` is optional.

File: src/manifest.ts [hunk 4]
```diff
@@ -9,3 +9,33 @@ export interface Moziot<T> {
...
+
+export interface WebthingsSettings {
+  exec: string;
+  strict_min_version: string;
+  strict_max_version: string;
+  primary_type: string;
+}
+
+export interface Options {
+  default: Record<string, unknown>;
```
Comment (mrstegeman) [inline src/manifest.ts:39]: `default` is optional.

File: src/manifest.ts [hunk 5]
```diff
@@ -9,3 +9,33 @@ export interface Moziot<T> {
   schema: Record<string, unknown>;
   config: T;
 }
+
+export interface ManifestFile {
+  manifest_version: number;
```
Comment (mrstegeman) [inline src/manifest.ts:14]: Missing:
* `content_scripts`
* `web_accessible_resources`

**Review by mrstegeman — CHANGES_REQUESTED** [review]

File: src/manifest.ts (hunk 1 above)
Comment (tim-hellhake) [reply src/manifest.ts]: done

File: src/manifest.ts (hunk 2 above)
Comment (tim-hellhake) [reply src/manifest.ts]: done

File: src/manifest.ts (hunk 4 above)
Comment (tim-hellhake) [reply src/manifest.ts]: done

File: src/manifest.ts (hunk 5 above)
Comment (tim-hellhake) [reply src/manifest.ts:14]: done

File: src/manifest.ts (hunk 3 above)
Comment (tim-hellhake) [reply src/manifest.ts]: done

**Review by mrstegeman — APPROVED** [review]

**Pull request merged**